src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -34,15 +34,16 @@ public String calculatePartition(final EventType eventType, final JSONObject eve
                     // The problem is that JSONObject doesn't override hashCode(). Therefore convert it to
                     // a string first and then use hashCode()
                     .map(pkf -> EventCategory.DATA.equals(eventType.getCategory()) ? DATA_PATH_PREFIX + pkf : pkf)
-                    .map(wrapFunction(okf -> traversableJsonEvent.get(okf).toString().hashCode()))
+                    .map(Try.wrap(okf -> traversableJsonEvent.get(okf).toString().hashCode()))
```

Comment:
I think it more flexible, and give us more opportunities. 
For example we could collect all exceptions instead of first one:
```java
List<Exception> exceptions = callableList.stream().map(callable -> Try.cons(callable)).filter(Try::isFailed).map(Try::getException).collect(Collections.toList());
```